photutils/psf/models.py

```diff
@@ -4,17 +4,428 @@
 from ..extern.nddata_compat import extract_array
 
+__all__ = ['FittableImageModel2D', 'NonNormalizable',
+           'IntegratedGaussianPRF', 'PRFAdapter',
+           'prepare_psf_model', 'get_grouped_psf_model']
 
-__all__ = ['IntegratedGaussianPRF', 'PRFAdapter', 'prepare_psf_model',
-           'get_grouped_psf_model']
+
+class NonNormalizable(Warning):
+    """
+    Used to undicate that a :py:class:`FittableImageModel2D` model is
+    non-normalizable.
+
+    """
+    pass
+
+
+class FittableImageModel2D(Fittable2DModel):
+    """
+    A fittable 2D model of an image allowing for image intensity scaling
+    and image translations.
+
+    This class takes 2D image data and computes the
+    values of the model at arbitrary locations (including at intra-pixel,
+    fractional positions) within this image using spline interpolation
+    provided by :py:class:`~scipy.interpolate.RectBivariateSpline`.
+
+    The fittable model provided by this class has three model parameters:
+    an image intensity scaling factor (`flux`) which is applied to
+    (normalized) image, and two positional parameters (`x_0` and `y_0`)
+    indicating the location of a feature in the coordinate grid on which
+    the model is to be evaluated.
+
+    If this class is initialized with `flux` (intensity scaling factor)
+    set to `None`, then `flux` is be estimated as ``|sum(data)|``.
+
+    Parameters
+    ----------
+    data : numpy.ndarray
+        Array containing 2D image.
+
+    origin : tuple, None, optional
+        A reference point in the input image ``data`` array. When origin is
+        `None`, origin will be set at the middle of the image array.
+
+        If `origin` represents the location of a feature (e.g., the position
+        of an intensity peak) in the input ``data``, then model parameters
+        `x_0` and `y_0` show the location of this peak in an another target
+        image to which this model was fitted. Fundamentally, it is the
+        coordinate in the model's image data that should map to
+        coordinate (`x_0`, `y_0`) of the output coordinate system on which the
+        model is evaluated.
+
+        Alternatively, when `origin` is set to ``(0,0)``, then model parameters
+        `x_0` and `y_0` are shifts by which model's image should be translated
+        in order to match a target image.
+
+    normalize : bool, optional
+        Indicates whether or not the model should be build on normalized
+        input image data. If true, then the normalization constant (*N*) is
+        computed so that
+
+        .. math::
+            N \cdot C \cdot |\Sigma_{i,j}D_{i,j}| = 1,
+
+        where *N* is the normalization constant, *C* is correction factor
+        given by the parameter ``correction_factor``, and :math:`D_{i,j}` are
+        the elements of the input image ``data`` array.
+
+    correction_factor : float, optional
+        A strictly positive number that represents correction that needs to
```

Comment:
> Finally, I do not want to have a very specific naming is not too specific, 

Regardless of the exact word it needs *something* to make it clear that it's about the overall flux/normalization/amplitude/whatever of the model.  `correction_factor` on its own could be lots of things: maybe it's a size-related correction factor?  Or an oversampling level? I only suggested `flux_correction` because the first words of the docstring are "represents correction that needs to be applied to model's `flux.", so that seemed internally consistent.  If you have a better word than flux I'm fine with that, but the name `correction_factor` doesn't imply any direct connection to the model's amplitude (which is the key element).